ts/src/main/scala/ch/epfl/ts/data/StrategyParameters.scala

```diff
@@ -8,42 +12,70 @@ import scala.reflect.ClassTag
  * Class behaving much like a map, designed to hold values
  * for a trading strategy's parameters.
  * Each parameter checks the validity of its value on instantiation.
- * 
+ *
  * It can be constructed with an arbitrary number of parameters.
  */
-class StrategyParameters(params: (String, Parameter[_])*) {
+class StrategyParameters(params: (String, Parameter)*) {
   type Key = String
   val parameters = params.toMap
-  
+
+  /**
+   * @return True only if the key is available in `parameters`
+   */
+  def has(key: Key): Boolean = parameters.contains(key)
+  /**
+   * @return True only if the key is available in `parameters`
+   *         AND it has the expected `Parameter` type
+   */
+  def hasWithType(key: Key, companion: ParameterTrait): Boolean =
+	  parameters.get(key) match {
+	    case Some(p) => (p.companion == companion)
+      case _ => false
+	  }
+
+  /**
+   * Get the value if it is there, otherwise throw an exception.
+   * Use it only if you are confident `params` contains the desired key.
+   */
+  def get[V: ClassTag](key: Key): V = {
+    parameters.get(key) match {
+      case Some(p) => p.value() match {
+        case v: V => v
+        case _ => throw new IndexOutOfBoundsException("Key `" + key + "` was not found.")
+      }
+      case _ => throw new IndexOutOfBoundsException("Key `" + key + "` was not found.")
+    }
+  }
+
   /**
    * Similar to what a map's `get` would do.
```

Comment:
I think we need this for optional parameters, when you're not sure the parameter is there and want to be able to fallback on a default value.